src/Azure.Functions.Cli/Extensions/ZipFileExtensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.IO;
+using Azure.Functions.Cli.Helpers;
+using Ionic.Zip;
+using Mono.Unix;
+
+namespace Azure.Functions.Cli.Extensions
+{
+    public static class ZipFileExtensions
+    {
+        public static void AddFile(this ZipFile file, string fileName, string zippedName, string zipRoot)
+        {
+            Stream openDelegate(string filename) => File.Open(fileName, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+            void closeDelegate(string filename, Stream stream) => stream.Close();
+
+            file.AddFile(zippedName, zipRoot, openDelegate, closeDelegate);
+        }
+
+        public static void AddFile(this ZipFile file, string fileName, string zipRoot, OpenDelegate openDelegate, CloseDelegate closeDelegate)
+        {
+            var entry = file.AddEntry(fileName.FixFileNameForZip(zipRoot), openDelegate, closeDelegate);
+
+            // If not Windows, we can assume it's a Unix environment
+            // For Unix environments, we need to explicity add file attributes in the Zipfile
+            if (!PlatformHelper.IsWindows)
+            {
+                if (TryGetUnixFileAttributes(fileName, out int fileAttributes))
```

Comment:
May need to get rid of this, as DotNetZip's ExternalAttributes are read differently. 
After several trail and error, I realized even with ExternalAttributes set properly such that permissions are propogated when unzipping, fuse-mount does not read it the same way as unzip would. 

Basically this is a bit tricky to make sure ExternalAttributes are read properly. I will try to find some other way.

Reply:
Do you know if executable bit is set correctly when using `zip(1)` on linux or mac?

Reply:
fuse has a bunch of options http://man7.org/linux/man-pages/man8/mount.fuse.8.html#OPTIONS that we can see if they make a difference. 

I'm trying to understand if this is a limitation of fuse-zip or of DotNetZip.

Reply:
`zip(1)` works fine, with fuse-zip understanding it correctly. I think I need to look a bit deeper into how fuse-zip retrieves file permissions. It's either a limitation of DotNetZip, or I need to figure out a way to use ExternalAttributes in a way that is read same by `unzip` as well as `fuse-zip`. I will look at how `zip(1)` does it.